internal/adminapi/client_test.go

```diff
@@ -0,0 +1,239 @@
+package adminapi
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"io/ioutil"
+	"math/big"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+	"time"
+)
+
+var ca *x509.Certificate
+var caPEM *bytes.Buffer
+var caPrivateKeyPEM *bytes.Buffer
+var certPEM *bytes.Buffer
+var certPrivateKeyPEM *bytes.Buffer
```

Comment:
I see we're using several globs in the tests which are effectively the product of the `buildTLS()` function, but using globs in this way could make the code a bit harder to understand and even harder to maintain in the future (e.g. if someone turned on parallelization of these tests, there would be immediate race conditions).

Suggestion: have `buildTLS()` return the TLS configurations it creates instead and provide more encapsulation for each individual test.